packages/api/src/task/scheduler.ts

```diff
@@ -378,18 +378,24 @@ export class TaskScheduler {
     userId?: string,
     requesterId?: string,
   ) {
+    const uId = inputAsset?.userId || outputAsset?.userId || userId;
+    const user = await db.user.get(uId, { useCache: true });
```

Comment:
I think the caller of this function always has the `User` object available on `req.user`. Shouldn't we just pass that down in the args instead of risking another DB round-trip here? They should like also send the `req.project`, or even send _only_ the `req.project.id` and avoid any repetition of the default project ID logic here.